@uportal/esco-content-menu-lit/src/portlet-card.ts

```diff
@@ -0,0 +1,194 @@
+/*Lit*/
+import { html, LitElement, css, unsafeCSS, TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { styleMap } from 'lit/directives/style-map.js';
+/*SCSS*/
+import PortletCardScss from '@styles/portlet-card.scss';
+/*Mixins*/
+import { LitLoggable } from '@mixins/litLoggable';
+/*Helpers*/
+import sizeHelper from '@helpers/sizeHelper';
+import pathHelper from '@helpers/pathHelper';
+import textHelper from '@helpers/textHelper';
+/*Components*/
+import './lit-ellipsis';
+import './action-favorites';
+
+@customElement('esco-portlet-card')
+export class PortletCard extends LitLoggable(LitElement) {
+  @property({ type: Array })
+  messages = [];
+  @property({ type: Object, attribute: 'portlet-desc' })
+  portletDesc: Portlet | undefined;
+  @property({ type: String, attribute: 'css-class' })
+  cssClass = 'portlet-card';
+  @property({ type: String, attribute: 'icon-background-color' })
+  iconBackgroundColor = 'transparent';
+  @property({
+    type: String,
+    hasChanged(newVal: string) {
+      return sizeHelper.validate(newVal, true, true);
+    },
+  })
+  size = 'medium';
+  @property({ type: String, attribute: 'card-hover-src' })
+  cardHoverSrc = 'none';
+  @property({ type: Boolean, attribute: 'background-is-dark' })
+  backgroundIsDark = false;
+  @property({ type: Boolean, attribute: 'is-favorite' })
+  isFavorite = false;
+  @property({ type: Boolean, attribute: 'hide-action' })
+  hideAction = false;
+  @property({ type: Boolean })
+  debug = false;
+
+  @state()
+  _dataIsParsed = false;
+
+  private _fname = '';
+  private _chanId = 0;
+  private _pTitle = '';
+  private _description = '';
+  private _portletTitle = '';
+  private _canFavorite = false;
+  private _iconUrl = '';
+
+  shouldUpdate(): boolean {
+    if (this.portletDesc !== undefined) return true;
+    this.errorLog(
+      'Attribute portlet-desc or property portletDesc must be defined'
+    );
+    return false;
+  }
+
+  willUpdate(changedProperties: Map<string | number | symbol, unknown>): void {
```

Comment:
Another area which could be fine here.
But may want to be discussed further.
Does it make sense to stick with lifecycle based design for components?

The hooks based model has taken both React and Vue over, and can be supported in Lit as well https://github.com/matthewp/haunted#readme